Pardot destination (#404)
rudderlabs/rudderstack-docs: destinations/marketing/pardot.mdx

</div>

## Identify

The `identify` call lets you capture the details of a visiting user along with any associated traits such as their name, email address, etc.

A sample `identify` call is shown below:

```javascript
rudderanalytics.identify("userId", {
  "active_seats": 6,
  "salutation": "Mr",
  "firstName": "Alex",
  "lastName": "Keener",
  "email": "alex@example.com",
  "website": "https://www.example.com",
  "score": 16,
  "company": "Example Organization",
  "job_title": "Sales Executive",
  "department": "Sales",
  "country": "USA",
  "address": {
    "street": "6649 N Blue Gum Street",
    "city": "New Orleans",
    "state": "LA",
    "zip": "00000"
  },
  "phone": "8005550100",
  "annual_revenue": "100000",
  "employees": "100",
  "industry": "Technology sector",
  "notes": "example notes",
  "address_two": "example address",
  "fax": "fax321",
  "years_in_business": 13,
  "comments": "good",
  "is_do_not_email": "false",
  "opted_out": "false",
  "is_reviewed": "false",

>   "is_starred": "true",

Author: done

  "territory": "territory",
  "campaign_id": 12345
}, {
  externalId: [{
    type: "crmfid", // can be pardotid for sending pardot ID
    id: "crmFidUser",
  }]
});
```
The following behavior can be observed:

> * An error is thrown if both the `email` and `crmfid`/`pardotid` fields are absent.

Reviewer: ```suggestion
* An error is thrown if both the `email` and `crmfid`/`pardotid` fields are absent.
```
[non-blocking]
@ishachopra0401 This line looks a bit confusing seems like `email` is also needed as externalId

`email` is taken from traits whereas `crmfid`/`pardotid` is taken from externalId

Either `email` or externalId (`crmfid`/`pardotid`) is required

Author: Updated, Thanks.

* If only `email` is present or Pardot already contains more than one prospect with the same email id, a new prospect is created as Pardot does not allow write operation on email ids.
* Only the first `externalId` is considered in case of multiple `externalIds`.

## FAQ

### How to find crmfid (used for upsert operation)?
You can find the crmfid (CRM Lead ID) under any **Prospect** by clicking on **Sync with CRM** as shown below:

<img src="../../assets/event-stream-destinations/Pardot-crmfid1.png" alt="Pardot CRM Lead ID"/>
The CRM Lead ID can be located here:
<img src="../../assets/event-stream-destinations/Pardot-crmfid2.png" alt="Pardot CRM Lead ID"/>

### How to update a prospect data?

To update the prospect data, you need to send a payload that contains `externalId` in the payload as shown below:
```javascript
// Sample payload with `id` data for update
{
  "externalId": [{
    "type": "pardotid"
    "id": "12345",
  }]
}
```
- `type` field can contain either of the below:
  - Pardot Id with field name as `pardotid`
  - CRM FID with field name as `crmfid`
- `id` field contains the actual value

## Contact us

> 

Reviewer: Currently, users can send data to Pardot’s production environment & not sandbox. Sandbox feature is will be released in the future - can we mention this in the FAQ

Author: Added to the FAQ, Thanks.

---
# slug: "/docs/destinations/marketing/pardot"
title: "Pardot"
description: Step-by-step guide to send your event data from RudderStack to Pardot.
---

# Pardot

> [Pardot](https://www.pardot.com/) is a Salesforce product which generates and nurtures the leads as well as empowers sales to close more deals. It is a marketing automation solution that enables organizations to track and measure the effectiveness of their communications, gain insight into user behavior, and personalize content across campaigns based on any specific criteria.

Reviewer 2: ```suggestion
[Pardot](https://www.pardot.com/) is a Salesforce product and a marketing automation solution that enables organizations to track and measure the effectiveness of their communications, gain insight into user behavior, and personalize content across campaigns based on any specific criteria.
```

Reviewer 2: minimized the intro a bit

| :------------------ | :------------ | :------------ | :------------ |
| **Device mode**     | -             | -             | -             |
| **Cloud mode**      | **Supported** | **Supported** | **Supported** |

<div class="infoBlock">

To know more about the difference between cloud mode and device mode in RudderStack, refer to the <a href="https://rudderstack.com/docs/connections/rudderstack-connection-modes/">RudderStack Connection Modes</a> guide.
</div>

Once you have confirmed that the source platform supports sending events to Pardot, follow these steps:

- From your [**RudderStack dashboard**](https://app.rudderstack.com/), add the source. Then, from the list of destinations, select **Pardot**.

<div class="infoBlock">

Follow our guide on <a href="https://rudderstack.com/docs/connections/adding-source-and-destination-rudderstack/">Adding a Source and Destination in RudderStack</a> for more information.
</div>

## Connection settings

To successfully configure Pardot as a destination, you will need to configure the following settings:

<img src="../../assets/event-stream-destinations/pardot-connection-settings3.png" alt="Pardot connection settings"/>

> - **Business Unit Id**: Your **Business Unit ID** is the 18 character long string starting with `0Uv`. You can find it in your **Salesforce** account by following the below steps:

Reviewer 2: ```suggestion
- **Business Unit Id**: Your **Business Unit ID** is the 18 character long string starting with `0Uv`. You can find it in your **Salesforce** account by following the steps listed below:
```

>   1. Go to **Setup** at the top right corner and click on **Pardot** option. 

Reviewer 2: ```suggestion
  1. Go to **Setup** at the top right corner and click on **Pardot**. 
```

>   2. Click on **Pardot Account Setup** to locate the Business Unit ID.

Reviewer 2: "to obtain the Business Unity ID" -- would this be a better alternative?

Reviewer 2: unless the ID is hidden and becomes visible only after clicking on Pardot Account Setup..

<img src="../../assets/event-stream-destinations/pardot-business-unit-id.png" alt="Pardot Business Unit Id"/>

> - **Campaign Id**: You can find the Campaign ID (or piCId) under **Marketing** > **Campaigns** in your Pardot account. After selecting the desired website campaign, press **View Tracking Code**.

Reviewer 2: ```suggestion
- **Campaign Id**: You can find the Campaign ID (or piCId) under **Marketing** > **Campaigns** in your Pardot account. After selecting the desired website campaign, click on **View Tracking Code**.
```

<div class="infoBlock">
  

> Alternatively, you can go to the <a href="https://pi.pardot.com/campaign">Pardot campaigns</a> and select the desired campaign from the list. The last 5 digits of the campaign URL is your campaign Id. For example, 12345 is the campaign Id for https://pi.pardot.com/campaign/read/id/12345 campaign URL.

Reviewer 2: ```suggestion
Alternatively, you can go to the <a href="https://pi.pardot.com/campaign">Pardot campaigns</a> and select the desired campaign from the list. The last 5 digits of the campaign URL is your campaign ID. For example, <code class="inline-code">12345</code> is the campaign ID for the campaign URL <code class="inline-code">https://pi.pardot.com/campaign/read/id/12345</code> .
```

A sample `identify` call is shown below:

```javascript
rudderanalytics.identify("userId", {
  "active_seats": 6,
  "salutation": "Mr",
  "firstName": "Alex",
  "lastName": "Keener",
  "email": "alex@example.com",
  "website": "https://www.example.com",
  "score": 16,
  "company": "Example Organization",
  "job_title": "Sales Executive",
  "department": "Sales",
  "country": "USA",
  "address": {
    "street": "6649 N Blue Gum Street",
    "city": "New Orleans",
    "state": "LA",
    "zip": "00000"
  },
  "phone": "8005550100",
  "annual_revenue": "100000",
  "employees": "100",
  "industry": "Technology sector",
  "notes": "example notes",
  "address_two": "example address",
  "fax": "fax321",
  "years_in_business": 13,
  "comments": "good",
  "is_do_not_email": false,
  "opted_out": false,
  "is_reviewed": false,
  "is_starred": true,
  "territory": "territory",
  "campaign_id": 12345
}, {
  externalId: [{
    type: "crmfid", // can be pardotid for sending pardot ID
    id: "crmFidUser",
  }]
});
```
The following behavior can be observed:

> * An error is thrown if either `email` or externalId (`crmfid`/`pardotid`) is absent.

Reviewer 2: ```suggestion
* RudderStack throws an error if either `email` or `externalId` (`crmfid`/`pardotid`) is absent.
```

> * An update API call is performed if `email` is absent but either `pardotid` or `crmfid` is present. However, if `pardotid` and `crmfid` are not already linked to a prospect, an error is thrown.

Reviewer 2: ```suggestion
* RudderStack calls the update API if `email` is absent but either `pardotid` or `crmfid` is present. However, if `pardotid` and `crmfid` are not already linked to a prospect, it throws an error.
```

* An error is thrown if `email` is absent and `prospectid` or `crmfid` is entered incorrectly.

> * If only `email` is present or Pardot already contains more than one prospect with the same email id, a new prospect is created as Pardot does not allow write operation on email ids.

Reviewer 2: ```suggestion
* If only `email` is present, or Pardot already contains more than one prospect with the same email ID, a new prospect is created as Pardot does not allow write operation on email IDs.
```

> * Only the first `externalId` is considered in case of multiple `externalIds`.

Reviewer 2: ```suggestion
* RudderStack considers only the first `externalId` in case of multiple `externalIds`.
```

## FAQ

> ### How to find crmfid (used for upsert operation)?

Reviewer 2: ```suggestion
### How do I find the crmfid (used for upsert operation)?
```

You can find the crmfid (CRM Lead ID) under any **Prospect** by clicking on **Sync with CRM** as shown below:

<img src="../../assets/event-stream-destinations/Pardot-crmfid1.png" alt="Pardot CRM Lead ID"/>
The CRM Lead ID can be located here:
<img src="../../assets/event-stream-destinations/Pardot-crmfid2.png" alt="Pardot CRM Lead ID"/>

> ### How to update a prospect data?

Reviewer 2: ```suggestion
### How do I update a prospect's data?
```

To update the prospect data, you need to send a payload that contains `externalId` in the payload as shown below:
```javascript
// Sample payload with `id` data for update
{
  "externalId": [{
    "type": "pardotid"
    "id": "12345",
  }]
}
```

> - `type` field can contain either of the below:

Reviewer 2: ```suggestion
- `type` can contain either of the following fields:
```

>   - Pardot Id with field name as `pardotid`

Reviewer 2: ```suggestion
  - Pardot ID with field name as `pardotid`
```

>   - CRM FID with field name as `crmfid`

Reviewer 2: we should be consistent with this term - above we say crmfid

- `id` field contains the actual value

> ### Can I send event data to Pardot's Sandbox environment?

Reviewer 2: ```suggestion
### Can I send the event data to Pardot's Sandbox environment?
```

> Unfortunately, you can currently send event data only to the Pardot's **Production** environment and not **Sandbox**.

Reviewer 2: ```suggestion
Unfortunately, you can currently send the event data only to the Pardot's **production** environment. Sandbox environments are not currently supported.
```
